SingularityService/src/main/java/com/hubspot/singularity/data/CuratorAsyncManager.java

```diff
@@ -217,6 +220,51 @@ public void processResult(CuratorFramework client, CuratorEvent event) throws Ex
     }
   }
 
+  private <T extends SingularityId> List<T> notExistsThrows(final String pathNameforLogs, final Map<String, T> pathsMap) throws Exception {
+    if (pathsMap.isEmpty()) {
+      return Collections.emptyList();
+    }
+
+    final List<T> objects = Lists.newArrayListWithCapacity(pathsMap.size());
+
+    final CountDownLatch latch = new CountDownLatch(pathsMap.size());
+
+    final BackgroundCallback callback = new BackgroundCallback() {
+
+      @Override
+      public void processResult(CuratorFramework client, CuratorEvent event) throws Exception {
+        if (event.getStat() == null) {
+          objects.add(pathsMap.get(event.getPath()));
+          latch.countDown();
```

Comment:
what's the point of these 2 statements? latch.countDown(); return; --> latch.countDown()